src/test/java/com/hocs/test/pages/decs/Search.java

```diff
@@ -244,546 +317,247 @@ public void clickEscalateComplaintsCaseToStage2() {
-            default:
-                pendingStep(criteria + " is not defined within " + getMethodName());
-        }
-    }
-
-    public void assertComplaintsInformationRandomSearchResult(String criteria) {
-        waitForResultsPage();
-        WebElementFacade cell = null;
-        String displayedValue = null;
-        String expectedValue = null;
+        List<WebElementFacade> listOfCasesWithValue;
         int numberOfCasesDisplayed = getNumberOfSearchResults();
-        int randomNumber = new Random().nextInt(numberOfCasesDisplayed) + 1;
-        WebElementFacade randomSearchResultHypertext = findBy("//tr[" + randomNumber + "]/td/a");
-        String randomSearchResult = randomSearchResultHypertext.getText();
-        String caseType = randomSearchResult.split("/")[0];
+        int numberOfUnallocatedCases = findAll("//td[3][not(text()='" + getCurrentUser().getUsername() + "')]/ancestor::tbody/tr/td/a").size();
+        int randomNumber = new Random().nextInt(numberOfUnallocatedCases) + 1;
+        WebElementFacade unallocatedRandomResultHypertext =
+                findBy("//td[3][not(text()='" + getCurrentUser().getUsername() + "')]/ancestor::tbody/tr[" + randomNumber + "]/td/a");
+        String randomSearchResult = unallocatedRandomResultHypertext.getText();
+        System.out.println("Random case selected for search verification: " + randomSearchResult);
         setSessionVariable("randomCaseRef").to(randomSearchResult);
-        switch (criteria.toUpperCase()) {
-            case "CASE TYPE":
-                expectedValue = sessionVariableCalled("searchCaseType");
-                List<WebElementFacade> listOfCaseRefs = findAll("//td[2]/a[contains(text(), '" + expectedValue + "/')]");
-                assertThat(numberOfCasesDisplayed == listOfCaseRefs.size(), is(true));
-                displayedValue = expectedValue;
-                break;
-            case "CORRESPONDENT FULL NAME":
-                if (!caseType.equalsIgnoreCase("IEDET")) {
-                    cell = findBy("//a[text()='" + randomSearchResult + "']/parent::td/preceding-sibling::td");
-                } else {
-                    safeClickOn(randomSearchResultHypertext);
-                    peopleTab.selectPeopleTab();
-                    cell = findBy("//th[text()='Name']/following-sibling::td");
-                }
-                expectedValue = sessionVariableCalled("searchCorrespondentFullName");
-                break;
-            case "CORRESPONDENT POSTCODE":
-                if (!caseType.equalsIgnoreCase("IEDET")) {
-                    if (!caseType.equalsIgnoreCase("SMC") && !caseType.equalsIgnoreCase("POGR")) {
-                        cell = findBy("//a[text()='" + randomSearchResult + "']/parent::td/following-sibling::td[4]");
+        waitForResultsPage();
+        if (criteria.equalsIgnoreCase("CASE TYPE")) {
+            searchFieldUsedInCaseTypes = Arrays.asList("DCU", "MPAM", "COMP", "IEDET", "SMC", "POGR", "BF", "FOI", "TO");
```

Comment:
I think this variable would be more intuitve if it was something like caseTypesSearchFieldIsAvailableFor